### ISSUE TITLE
new-haddock can't produce documentation suitable for Hackage?

### ISSUE THREAD

**Issue by RyanGlScott** [haskell/cabal#5341]
I'm not sure if this is expected behavior or not, but it's certainly throwing me for a loop. I occasionally need to upload Haddocks to Hackage manually. With old-style `cabal`, my usual workflow would be:

1. Create a sandbox.
2. Install all dependencies with `--enable-documentation` on.
3. Run `cabal haddock --for-hackage`.
4. Run `cabal upload -d <doc-tarball> --publish`

I'm trying to figure out how to emulate this workflow with `new-haddock`, but it doesn't seem to be quite up to the task. My first inclination was to try this:

```
$ cabal new-build --enable-documentation
```

However, this did not build my dependencies with documentation enabled. It simply built the package itself (using previously built versions of the dependencies, without documentation) and ran Haddock on the package, and unsurprisingly, that resulted in unbound link destinations:

```
Warning: Data.Eliminator.TH: could not find link destinations for:
    ~> Sing @@
Warning: Data.Eliminator: could not find link destinations for:
    ~> Sing @@ Nat Z S
Warning: Data.Eliminator.TypeNats: could not find link destinations for:
    ~> Sing @@
```

Passing `--haddock-for-hackage` didn't seem to help either.

I also tried directly invoking `cabal new-haddock --haddock-for-hackage`, but that too did not build the dependencies with documentation enabled.

**Comment by 23Skidoo** [comment]
I think `new-haddock` should support this. Most likely just unfinished.

**Comment by merijn** [comment]
I think this works in HEAD or one of Alex's branches, since I've successfully build and uploaded Hackage docs using new-haddock 1 or 2 months ago when he was working on that.

@RyanGlScott are you using cabal-install HEAD or older?

**Comment by RyanGlScott** [comment]
@merijn, I used `cabal-install-2.2` in my original attempt. I just tried to use the latest `cabal-install` HEAD from @hvr's Ubuntu PPA, but I experience the same issues.

**Comment by RyanGlScott** [comment]
I'd say that this is fixed after #5395, since I can now run:

```
$ cabal new-build --enable-documentation --haddock-for-hackage
```

And it will build suitable-for-Hackage Haddocks, building dependencies with documentation enabled if need be.
